odbc/src/api/functions.rs

```diff
@@ -1785,11 +1785,441 @@ pub unsafe extern "C" fn SQLGetInfo(
 #[no_mangle]
 pub unsafe extern "C" fn SQLGetInfoW(
     connection_handle: HDbc,
-    _info_type: InfoType,
-    _info_value_ptr: Pointer,
-    _buffer_length: SmallInt,
-    _string_length_ptr: *mut SmallInt,
+    info_type: InfoType,
+    info_value_ptr: Pointer,
+    buffer_length: SmallInt,
+    string_length_ptr: *mut SmallInt,
+) -> SqlReturn {
+    panic_safe_exec!(
+        || sql_get_infow_helper(
+            connection_handle,
+            info_type,
+            info_value_ptr,
+            buffer_length,
+            string_length_ptr
+        ),
+        connection_handle
+    )
+}
+
+unsafe fn sql_get_infow_helper(
+    connection_handle: HDbc,
+    info_type: InfoType,
+    info_value_ptr: Pointer,
+    buffer_length: SmallInt,
+    string_length_ptr: *mut SmallInt,
 ) -> SqlReturn {
+    match info_type {
+        // SQL_DRIVER_NAME
+        InfoType::DriverName => {
+            // A character string with the file name of the driver used to access the data source.
+            // TODO: (code review) What file name should be used here? Is this something that
+            //       will need to be dynamically determined or is it something we can hardcode? 
+        }
+        // SQL_DRIVER_VER
+        InfoType::DriverVer => {
+            // TODO: (code review) is it ok this isn't of the form ##.##.####?
+            // The driver version can be obtained from the Cargo.toml file.
+            // The env! macro call below gets the version from the Cargo file
+            // at compile time.
+            let version = env!("CARGO_PKG_VERSION");
+            return i16_len::set_output_wstring(
+                version,
+                info_value_ptr as *mut WChar,
+                buffer_length as usize,
+                string_length_ptr,
+            );
+        }
+        // SQL_DRIVER_ODBC_VER
+        InfoType::DriverOdbcVer => {
+            // TODO: (code review) should we use env.odbc_version for this?
+        }
+        // SQL_SEARCH_PATTERN_ESCAPE
+        InfoType::SearchPatternEscape => {
+            // This value does not apply to LIKE, which is the only operator/function
+            // MongoSQL supports that could accept a search pattern. Therefore, the
+            // empty string is always returned.
+            return i16_len::set_output_wstring(
+                "",
```

Comment:
Hmm, I'm not sure what the docs mean when they say:
> This InfoType is limited to catalog functions. For a description of the use of the escape character in search pattern strings, see [Pattern Value Arguments](https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/pattern-value-arguments?view=sql-server-ver16).

My interpretation of "catalog functions" here was functions that appear _within_ SQL queries. You are suggesting this applies to the ODBC functions just as "list tables" and "list columns" etc. It's confusing since the docs refer to the `LIKE` operator (which appears in a SQL query, not in the ODBC listing functions). Can you clarify how this InfoType is used?

Reply:
Yes, this time, I am sure that catalog functions are SQLTables,  SQLColumns, etc... :)
Even, better, this is in the spec 🎉 
See [here](https://learn.microsoft.com/en-us/sql/odbc/reference/develop-app/catalog-functions-in-odbc)
Just for future reference, it might help to remember what they mean by catalog functions : Catalog in ODBC is generally similar to DatabaseMetadata in JDBC. It's everything that is useful for a user to know the DB structure/data model.

Reply:
> Catalog in ODBC is generally similar to DatabaseMetadata in JDBC. It's everything that is useful for a user to know the DB structure/data model.

This is super useful for future reference, thank you!

Reply:
updated the comment